### PR TITLE
task docs

### PR DESCRIPTION
**Pull request by kennyworkman**

### CODE REVIEW

File: docs/source/basics/when_to_create_task.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+A workflow is an analysis that takes in some input, processes it in one or more
+steps and produces some output.
+
+---
+
+Reasons to create a new task:
+
+* It is easier to manage dependencies / containers for smaller tasks.
+* Tasks can be reused between different workflows.
+* Tasks can be assigned different computing resources.
```
Comment (AidanAbd) [inline docs/source/basics/when_to_create_task.md:12]:
```suggestion
* Each task can be assigned different computing resources.
```

File: docs/source/basics/when_to_create_task.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+* It is easier to isolate and debug failed tasks if tasks are small
+* It is easier to retry workflows from the last failed task if tasks are small.
+The last succeeded task will be "further along" in the workflow.
```
Comment (AidanAbd) [inline docs/source/basics/when_to_create_task.md:15]:
```suggestion
* It is easier to retry workflows from the last failed task if the tasks are small.
The last succeeded task will be "further along" in the workflow.
```

**Review by AidanAbd — CHANGES_REQUESTED** [review]
This review includes all grammarly suggestions and some stuff I rewrote because it did not initially make sense to me.

File: docs/source/basics/when_to_create_task.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# When to create a Task?
+
+A workflow is an analysis that takes in some input, processes it in one or more
+steps and produces some output.
+
+---
+
+Reasons to create a new task:
+
+* It is easier to manage dependencies / containers for smaller tasks.
```
[the remark this answers is not in the document]
Comment (kennyworkman) [reply docs/source/basics/when_to_create_task.md:10]: responsibilities is ambiguous to me? i still think smaller is better

[the remark this answers is not in the document]
Comment (kennyworkman) [reply docs/source/basics/when_to_create_task.md:10]: actually `tasks using less code`

File: docs/source/basics/when_to_create_task.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# When to split up a task into multiple tasks
+
+When building a workflow with multiple tasks, it can be difficult to decide
+when to split larger tasks into smaller tasks. Some of the tradeoffs are listed
```
Comment (AidanAbd) [inline docs/source/basics/when_to_create_task.md:4]:
```suggestion
when to split larger tasks into multiple smaller tasks. Some of the tradeoffs are listed
```

**Review by AidanAbd — APPROVED** [review]
Two small things left

**Pull request merged**